src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/TaskQuerySpecification.java

```diff
@@ -1,63 +1,167 @@
+    private static final String TASK_ID = "taskId";
+    private static final String TASK_TYPE = "taskType";
+    private static final String ASSIGNEE = "assignee";
+    private static final String CASE_ID = "caseId";
+    private static final String JURISDICTION = "jurisdiction";
+    private static final String WORK_TYPE = "workTypeResource";
+    private static final String WORK_TYPE_ID = "id";
+
 
     private TaskQuerySpecification() {
         // avoid creating object
     }
 
     public static Specification<TaskResource> searchByState(List<CFTTaskState> cftTaskStates) {
-        return (root, query, builder) -> builder.in(root.get(STATE))
-            .value(cftTaskStates);
+        if (!CollectionUtils.isEmpty(cftTaskStates)) {
+            return (root, query, builder) -> builder.in(root.get(STATE))
+                .value(cftTaskStates);
+        }
+        return (root, query, builder) -> builder.conjunction();
     }
 
     public static Specification<TaskResource> searchByJurisdiction(List<String> jurisdictions) {
-        return (root, query, builder) -> builder.in(root.get(JURISDICTION))
-            .value(jurisdictions);
+        if (!CollectionUtils.isEmpty(jurisdictions)) {
+            return (root, query, builder) -> builder.in(root.get(JURISDICTION))
+                .value(jurisdictions);
+        }
+        return (root, query, builder) -> builder.conjunction();
     }
 
     public static Specification<TaskResource> searchByLocation(List<String> locations) {
-        return (root, query, builder) -> builder.in(root.get(LOCATION))
-            .value(locations);
+        if (!CollectionUtils.isEmpty(locations)) {
+            return (root, query, builder) -> builder.in(root.get(LOCATION))
+                .value(locations);
+        }
+        return (root, query, builder) -> builder.conjunction();
+    }
+
+    public static Specification<TaskResource> searchByCaseId(String caseId) {
+        return (root, query, builder) -> builder.equal(root.get(CASE_ID), caseId);
     }
 
-    public static Specification<TaskResource> searchByCaseId(List<String> caseIds) {
-        return (root, query, builder) -> builder.in(root.get(CASE_ID))
-            .value(caseIds);
+    public static Specification<TaskResource> searchByCaseIds(List<String> caseIds) {
+        if (!CollectionUtils.isEmpty(caseIds)) {
+            return (root, query, builder) -> builder.in(root.get(CASE_ID))
+                .value(caseIds);
+        }
+        return (root, query, builder) -> builder.conjunction();
     }
 
     public static Specification<TaskResource> searchByUser(List<String> users) {
-        return (root, query, builder) -> builder.in(root.get(ASSIGNEE))
-            .value(users);
+        if (!CollectionUtils.isEmpty(users)) {
+            return (root, query, builder) -> builder.in(root.get(ASSIGNEE))
+                .value(users);
+        }
+        return (root, query, builder) -> builder.conjunction();
```

Comment:
personal preference would be to use it as a safe guard: what I mean:

instead of negating the CollectionUtils.isEmpty(users)

```suggestion
        if (CollectionUtils.isEmpty(users)) {
          return (root, query, builder) -> builder.conjunction();
        }
       return (root, query, builder) -> builder.in(root.get(ASSIGNEE))
                .value(users);
```

this way the logic is always default

Reply:
With PMD check few times I see PMD warns to use negated condition first.